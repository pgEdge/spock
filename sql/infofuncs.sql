DO $$
BEGIN
	IF (SELECT setting::integer/100 FROM pg_settings WHERE name = 'server_version_num') = 904 THEN
		CREATE EXTENSION IF NOT EXISTS spock_origin;
	END IF;
END;$$;
CREATE EXTENSION spock;

SELECT spock.spock_max_proto_version();

SELECT spock.spock_min_proto_version();

-- test extension version
SELECT spock.spock_version() = extversion
FROM pg_extension
WHERE extname = 'spock';

DROP EXTENSION spock;

-- test upgrades
DO $$
BEGIN
        IF version() ~ 'Postgres-XL' THEN
                CREATE EXTENSION IF NOT EXISTS spock;
        ELSE
                CREATE EXTENSION IF NOT EXISTS spock VERSION '4.0.10';
        END IF;
END;
$$;
ALTER EXTENSION spock UPDATE;

SELECT spock.spock_version() = extversion
FROM pg_extension
WHERE extname = 'spock';

DROP EXTENSION spock;
