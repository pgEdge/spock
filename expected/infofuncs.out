DO $$
BEGIN
	IF (SELECT setting::integer/100 FROM pg_settings WHERE name = 'server_version_num') = 904 THEN
		CREATE EXTENSION IF NOT EXISTS spock_origin;
	END IF;
END;$$;
CREATE EXTENSION spock;
SELECT spock.spock_max_proto_version();
 spock_max_proto_version 
-------------------------
                       2
(1 row)

SELECT spock.spock_min_proto_version();
 spock_min_proto_version 
-------------------------
                       2
(1 row)

-- test extension version
SELECT spock.spock_version() = extversion
FROM pg_extension
WHERE extname = 'spock';
 ?column? 
----------
 t
(1 row)

DROP EXTENSION spock;
-- test upgrades
DO $$
BEGIN
        IF version() ~ 'Postgres-XL' THEN
                CREATE EXTENSION IF NOT EXISTS spock;
        ELSE
                CREATE EXTENSION IF NOT EXISTS spock VERSION '4.0.10';
        END IF;
END;
$$;
ALTER EXTENSION spock UPDATE;
SELECT spock.spock_version() = extversion
FROM pg_extension
WHERE extname = 'spock';
 ?column? 
----------
 t
(1 row)

DROP EXTENSION spock;
